packages/now-node/src/helpers.ts

```diff
@@ -126,27 +112,44 @@ export function sendError(
   res.end();
 }
 
-export function addHelpers(listener: NowListener): RequestListener {
-  return async function(_req, _res) {
-    const req = _req as NowRequest;
-    const res = _res as NowResponse;
-
+export function createServerWithHelpers(
+  listener: (req: NowRequest, res: NowResponse) => void | Promise<void>,
+  bridge: Bridge
+) {
+  async function wrappedListener(req: NowRequest, res: NowResponse) {
     try {
+      const reqId = req.headers['x-now-bridge-request-id'];
+
+      // don't expose this header to the client
+      delete req.headers['x-now-bridge-request-id'];
+
+      if (typeof reqId !== 'string') {
+        throw new ApiError(500, 'Internal Server Error');
+      }
+
+      const event = bridge.consumeEvent(reqId);
+
       req.cookies = parseCookies(req.headers.cookie || '');
       req.query = parseQuery(req);
-      req.body = await parseBody(req);
+      req.body = parseBody(req, event.body);
 
       res.status = statusCode => sendStatusCode(res, statusCode);
       res.send = data => sendData(res, data);
       res.json = data => sendJson(res, data);
 
-      listener(req, res);
+      await listener(req, res);
     } catch (err) {
       if (err instanceof ApiError) {
         sendError(res, err.statusCode, err.message);
       } else {
         throw err;
       }
     }
-  };
+  }
+
+  const server = new Server((req, res) => {
+    wrappedListener(req as NowRequest, res as NowResponse);
```

Comment:
```suggestion
    await wrappedListener(req as NowRequest, res as NowResponse);
```

I think you need an await here